src/TesApi.Web/Management/BatchQuotaVerifier.cs

```diff
@@ -106,30 +107,40 @@ public async Task CheckBatchAccountQuotasAsync(VirtualMachineInformation virtual
 
         if (needPoolOrJobQuotaCheck)
         {
-            if (batchUtilization.ActiveJobsCount + 1 > batchVmFamilyBatchQuotas.ActiveJobAndJobScheduleQuota)
-            {
-                throw new AzureBatchQuotaMaxedOutException($"No remaining active jobs quota available. There are {batchUtilization.ActivePoolsCount} active jobs out of {batchVmFamilyBatchQuotas.ActiveJobAndJobScheduleQuota}.");
-            }
+            var (_, exception) = CheckBatchAccountPoolOrJobQuotasImpl(required: 1, batchUtilization.ActiveJobsCount, batchUtilization.ActivePoolsCount, batchVmFamilyBatchQuotas.ActiveJobAndJobScheduleQuota, batchVmFamilyBatchQuotas.PoolQuota);
 
-            if (batchUtilization.ActivePoolsCount + 1 > batchVmFamilyBatchQuotas.PoolQuota)
+            if (exception is not null)
```

Comment:
The caller will throw the returned exception for every task assigned to every doomed new pool request (so could be lots). It's much more efficient to return the exception to selectively throw it for a subset of tasks rather than catch it to selectively rethrow it for that same subset of tasks.

The NET standard libraries do this "all the time"